data-semantics.md

```diff
@@ -0,0 +1,106 @@
+
+
+### Span Structure
+
+TODO
+
+
+
+## Semantic Tag Use-Cases
+
+The following tags are recommended for instrumentors who are trying to represent a particular type of data. Tag names follow a general structure of namespacing; the values are determined purely on a tag-by-tag basis.
+
+Some tags mentioned below may contain values of significant size. Handling of such values is implementation-specific: it is the responsibility of the Tracer to honor, drop, or truncate these tags as appropriate. However, care should be exercised on the part of the instrumentor, as even the generation or passing of such values to the Tracer may create undesirable overhead for the application.
+
+It is not required that all specified tags be used, even if one is used.
+
+
+### HTTP Server Tags
+
+These tags are recommended for spans marking entry into a HTTP-based service.
+
+* `http.url` - string
+    - URL of the request being handled in this segment of the trace, in [standard URI format](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier), excepting query string parameters.
+    - Protocol optional
+    - Examples:
+        - `https://domain.net/path/to?resource=here`
+        - `domain.net/path/to/resource`
+        - `http://user:pass@domain.org:8888`
+* `http.query` - string
+    - Query string parameters of the request being handled.
+    - Note that the preceding `?` character is not expected.
+    - Examples:
+        - `key=val&key2=val2`
+* `http.method` - string
+    - HTTP method of the request being handled.
+    - Case-insensitive
+    - Examples:
+        - `GET`, `POST`, `HEAD`
+* `http.status` - integer
+    - HTTP status code to be returned with HTTP response.
+    - Examples:
+        - `200`, `503`
+* `span.kind` - string
+    - Value of `s` should be used to indicate that this is a server side span (see "Peer Tags")
+
+
+### Peer Tags
+
+These tags can be provided by either client-side or server-side to describe the downstream (client case) or upstream (server case) peer being communicated with.
+
+* `peer.hostname` - string
+    - Remote hostname
+* `peer.ipv4` - string
+    - Remote IP v4 address
+* `peer.ipv6`
+    - Remote IP v6 address
+* `peer.port`
+    - Remote port
+* `peer.service` - string
+    - Remote service name
+* `span.kind` - string
+    - One of `c` or `s`, indicating if this span represents a client or server
+
+
+
+## Semantic Log Use-Cases
+
+Some events are point-in-time and do not apply to an entire span; these are reported via logged events.  For instance, an error or exception may be reported via a `log` call.
+
+
+### Exception
+
+* Event: `exception`
+* Payload:
+    - `type` - string
+        - Low-cardinality class of exception
+    - `val` - string
+        - Details associated with particular instance of this exception
+    - `tb` - string
+        - Stack trace provided at exception time
```

Comment:
Traceback, which is a python-ism for a stack trace.  I don't know that these are optimal--I stole the ones used in the python implementation, and they're pretty terse.

How about these instead: `type`, `message`, and `backtrace`?